### PR TITLE
[8.x] Compatible to phpunit 10

### PR DESCRIPTION
**Pull request by MortezaPoussaneh**
### Make tests compatible to phpunit 10 and add some small refactors
Base on phpunit document [setMethods](https://github.com/sebastianbergmann/phpunit/pull/3687) and [at](https://github.com/sebastianbergmann/phpunit/issues/4297) will be removed. I removed these methods so it can be easily possible to use phpunit 10.

### CODE REVIEW

File: tests/Database/DatabaseEloquentFactoryTest.php [hunk 1]
```diff
@@ -242,16 +238,11 @@ public function test_morph_to_relationship()
...
-                            }),
-                            ['admin' => 'Y'],
-                            'roles'
-                        )
-                        ->create();
+        $roleFactory = FactoryTestRoleFactory::times(3)->afterCreating(function ($role, $user) {
+            $_SERVER['__test.role.creating-role'] = $role;
+            $_SERVER['__test.role.creating-user'] = $user;
+        });
+        FactoryTestUserFactory::times(3)->hasAttached($roleFactory, ['admin' => 'Y'], 'roles')->create();
```
Comment (driesvints) [inline tests/Database/DatabaseEloquentFactoryTest.php:245]: Revert the code style changes you made here and only focus on the PHPUnit fixes.

File: tests/Database/DatabaseEloquentFactoryTest.php [hunk 2]
```diff
@@ -182,23 +182,19 @@ public function test_after_creating_and_making_callbacks_are_called()
...
+                $_SERVER['__test.post.state-user'] = $user;
+
+                return [];
+            })
+            // Test parents passed to callback...
+            ->afterCreating(function ($post, $user) {
+                $_SERVER['__test.post.creating-post'] = $post;
+                $_SERVER['__test.post.creating-user'] = $user;
+            });
+        FactoryTestUserFactory::times(10)->has($postFactory, 'posts')->create();
```
Comment (driesvints) [inline tests/Database/DatabaseEloquentFactoryTest.php:197]: Revert the code style changes you made here and only focus on the PHPUnit fixes.

File: tests/Database/DatabaseConnectionTest.php [hunk 3]
```diff
@@ -148,18 +151,42 @@ public function testTransactionLevelNotIncrementedOnTransactionException()
...
+        $exception = new ErrorException('server has gone away');
+        $pdo = $this->createMock(DatabaseConnectionTestMockPDO::class);
+        $requiredMethods = ['causedByLostConnection', 'reconnect'];
+        $connection = $this->getMockConnection($requiredMethods, $pdo);
+        $connection->expects($this->once())->method('causedByLostConnection')->with($exception)->willReturn(true);
+        $connection->expects($this->once())->method('reconnect');
+        $pdo->expects($this->once())->method('beginTransaction');
+        $this->privateCall($connection, 'handleBeginTransactionException', [$exception]);
+    }
+
```
Comment (driesvints) [inline tests/Database/DatabaseConnectionTest.php:189]: You seem to have added new tests here. These are probably best done in a separate PR.

File: tests/Auth/AuthGuardTest.php [hunk 4]
```diff
@@ -521,6 +520,6 @@ protected function getMocks()
 
     protected function getCookieJar()
     {
-        return new CookieJar(Request::create('/foo', 'GET'), m::mock(Encrypter::class), ['domain' => 'foo.com', 'path' => '/', 'secure' => false, 'httpOnly' => false]);
+        return new CookieJar();
```
Comment (driesvints) [inline tests/Auth/AuthGuardTest.php:523]: Why did you remove the arguments here?

File: tests/Foundation/FoundationAuthenticationTest.php [hunk 5]
```diff
@@ -16,7 +16,7 @@ class FoundationAuthenticationTest extends TestCase
     use InteractsWithAuthentication;
 
     /**
-     * @var \Mockery
+     * @var m
```
Comment (driesvints) [inline tests/Foundation/FoundationAuthenticationTest.php:19]: Revert this change.

File: tests/Foundation/FoundationAuthenticationTest.php [hunk 6]
```diff
@@ -29,7 +29,7 @@ class FoundationAuthenticationTest extends TestCase
     ];
 
     /**
-     * @var \Mockery
+     * @return Guard|m\LegacyMockInterface|m\MockInterface
```
Comment (driesvints) [inline tests/Foundation/FoundationAuthenticationTest.php:32]: Use fully qualified namespaces here.

File: tests/ReflectionHelpers.php [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+     * @throws ReflectionException
+     */
+    protected function setPrivateProperty($object, $property, $value)
+    {
+        $reflection = new ReflectionClass($object);
+        $reflectionProperty = $reflection->getProperty($property);
+        $reflectionProperty->setAccessible(true);
+        $reflectionProperty->setValue($object, $value);
+    }
+}
```
Comment (driesvints) [inline tests/ReflectionHelpers.php:55]: This class probably best belongs in a new PR alongside the new tests. Please remove this from this PR.

Also: a class like this probably belongs in the `Illuminate\Testing` component.

Comment (driesvints) [reply tests/ReflectionHelpers.php:55]: The docblocks here also don't match the rest of the framework's docblocks style and formatting.

File: tests/View/Blade/BladeCustomTest.php [hunk 8]
```diff
@@ -52,20 +55,17 @@ public function testCustomShortStatements()
         $this->assertEquals($expected, $this->compiler->compileString($string));
     }
 
-    public function testValidCustomNames()
+    /**
+     * @dataProvider customNamesDataProvider
+     * @param string $name
+     * @throws \ReflectionException
+     */
```
Comment (driesvints) [inline tests/View/Blade/BladeCustomTest.php:62]: Format this DocBlock as detailed in the contribution guide: https://laravel.com/docs/8.x/contributions#phpdoc

**Review by driesvints — CHANGES_REQUESTED** [review]
Great work on this PR, thanks! I went over it and highlighted some things.

File: tests/Auth/AuthGuardTest.php (hunk 4 above)
Comment (MortezaPoussaneh) [reply tests/Auth/AuthGuardTest.php:523]: :thinking: There wasn't any usage of them! I'm going to revert this change.

File: tests/Database/DatabaseConnectionTest.php (hunk 3 above)
Comment (MortezaPoussaneh) [reply tests/Database/DatabaseConnectionTest.php:189]: They are because by removing `at` function there was a method which was not covered anymore. These tests are covering that method.Do you think they should be in separate PR?

Comment (driesvints) [reply tests/Database/DatabaseConnectionTest.php:189]: Yeah I think it'll be more clear to handle these in a separate PR with a good explanation why it's needed. Thanks

Comment (MortezaPoussaneh) [reply tests/Database/DatabaseConnectionTest.php:189]: Ok, i will try it.

File: tests/Database/DatabaseConnectionTest.php [hunk 9]
```diff
@@ -148,14 +147,12 @@ public function testTransactionLevelNotIncrementedOnTransactionException()
...
-        $pdo->expects($this->at(0))
-            ->method('beginTransaction')
-            ->will($this->throwException(new ErrorException('server has gone away')));
-        $connection = $this->getMockConnection(['reconnect'], $pdo);
-        $connection->expects($this->once())->method('reconnect');
+        $connection = $this->getMockConnection(['createTransaction', 'fireConnectionEvent'], $pdo);
+        $connection->expects($this->once())->method('createTransaction');
+        $connection->expects($this->once())->method('fireConnectionEvent')->with('beganTransaction');
         $connection->beginTransaction();
         $this->assertEquals(1, $connection->transactionLevel());
```
Comment (driesvints) [inline tests/Database/DatabaseConnectionTest.php:157]: This test has become an entirely different one. I think it's best that we revert this part as well and try to tackle it together with those other tests?

File: tests/Database/DatabaseEloquentFactoryTest.php [hunk 10]
```diff
@@ -183,22 +183,22 @@ public function test_after_creating_and_making_callbacks_are_called()
...
+                        return [];
+                    })
+                    // Test parents passed to callback...
+                    ->afterCreating(function ($post, $user) {
+                        $_SERVER['__test.post.creating-post'] = $post;
+                        $_SERVER['__test.post.creating-user'] = $user;
+                    }),
+                'posts'
+            )
+            ->create();
```
Comment (driesvints) [inline tests/Database/DatabaseEloquentFactoryTest.php:201]: Revert the indentation here.

File: tests/Database/DatabaseEloquentFactoryTest.php [hunk 11]
```diff
@@ -243,15 +243,15 @@ public function test_morph_to_relationship()
...
-                        ->create();
+            ->hasAttached(
+                FactoryTestRoleFactory::times(3)->afterCreating(function ($role, $user) {
+                    $_SERVER['__test.role.creating-role'] = $role;
+                    $_SERVER['__test.role.creating-user'] = $user;
+                }),
+                ['admin' => 'Y'],
+                'roles'
+            )
+            ->create();
```
Comment (driesvints) [inline tests/Database/DatabaseEloquentFactoryTest.php:254]: Revert the indentation here.

File: tests/Foundation/FoundationAuthenticationTest.php [hunk 12]
```diff
@@ -29,7 +29,7 @@ class FoundationAuthenticationTest extends TestCase
     ];
 
     /**
-     * @var \Mockery
+     * @return Guard|\Mockery\LegacyMockInterface|\Mockery\MockInterface
```
Comment (driesvints) [inline tests/Foundation/FoundationAuthenticationTest.php:32]: The `Guard` class here still needs a FQN.

**Review by driesvints — CHANGES_REQUESTED** [review]
Cool. Just a couple more things.

File: tests/Foundation/FoundationAuthenticationTest.php (hunk 12 above)
Comment (MortezaPoussaneh) [reply tests/Foundation/FoundationAuthenticationTest.php:32]: Just for learning, why it should be FQN?

Comment (driesvints) [reply tests/Foundation/FoundationAuthenticationTest.php:32]: It's part of our coding standards.

Comment (MortezaPoussaneh) [reply tests/Foundation/FoundationAuthenticationTest.php]: Do you know why?! Is it laravel standard or is coming from somewhere else?! just curious :smile:

File: tests/Database/DatabaseConnectionTest.php (hunk 9 above)
Comment (MortezaPoussaneh) [reply tests/Database/DatabaseConnectionTest.php:157]: I have no idea how to fix this one without these changes! I'm removing `at` here, but original code is trying to throw exception in first call and do regular job in second call. Any idea?

Comment (driesvints) [reply tests/Database/DatabaseConnectionTest.php:157]: Let's just revert this for now and open a new PR later on where we can discuss this 👍

Comment (MortezaPoussaneh) [reply tests/Database/DatabaseConnectionTest.php:157]: :thinking: but whole purpose of PR will be failed. 
OK, i will revert it and i will change it with those two tests to cover what should be covered in first place.

**Review by driesvints — APPROVED** [review]
Awesome work 👍

**Review by GrahamCampbell — APPROVED** [review]
